src/Payments.sol

```diff
@@ -313,6 +313,28 @@ contract Payments is
             });
     }
 
+    /// @notice Checks if a rail is in debt by settling the payer's account lockup and checking if it's behind.
+    /// @param railId The ID of the rail to check.
+    /// @return True if the rail is in debt (payer's lockup cannot be fully settled to current epoch), false otherwise.
+    function isRailInDebt(
+        uint256 railId
+    ) external validateRailActive(railId) nonReentrant returns (bool) {
+        Rail storage rail = rails[railId];
+        
+        // Zero rate rails with empty rate change queues never contribute to debt
+        if (rail.paymentRate == 0 && rail.rateChangeQueue.isEmpty()) {
+            return false;
+        }
+        
+        Account storage payerAccount = accounts[rail.token][rail.from];
+        
+        // Settle the account lockup and get the epoch it was settled to
+        uint256 settledUpToEpoch = settleAccountLockup(payerAccount);
```

Comment:
If we do want to keep this PR then I don't think we want to actually settle, I think we want to follow the approach of #99.

I think we're better off either just leaving the solution in #99 for the operator to call, or if we really want an onchain method using rail id we can replace all this with 

```solidity
Rail storage rail = rails[railId];
(uint256 fundedUntilEpoch,,,) =  getAccountInfoIfSettled(rail.token, rail.from);
return fundedUntilEpoch < currentEpoch
```